### PR TITLE
Dropout module

### PR DESCRIPTION
**Pull request by franberchez**
Development of a new module called `dropout`.

This new module includes a hybrid version of dropout called `HybridDropout`, which mixes the standard dropout with the ordinal information of a dataset.

### CODE REVIEW

File: dlordinal/dropout/hybrid_dropout.py [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+
+        for module in self.model.modules():
+            if isinstance(module, HybridDropout):
+                module.batch_targets = targets
+
+
+class HybridDropout(nn.Module):
+    """Implements a Hybrid dropout methodology by :footcite:t:`berchez2024fusion` which
+    mix a standard dropout with an ordinal dropout. The ordinal dropout is based on the
+    correlation between the activation values of the of the neuron and the target labels
```
Comment (victormvy) [inline dlordinal/dropout/hybrid_dropout.py:67]: There is a type here
